service/controller/resource/nodepool/resource.go

```diff
@@ -53,3 +64,23 @@ func New(config Config) (*Resource, error) {
 func (r *Resource) Name() string {
 	return Name
 }
+
+func (r *Resource) getTenantClusterK8sClient(ctx context.Context, cluster *capiv1alpha3.Cluster) (k8sclient.Interface, error) {
+	var k8sClient k8sclient.Interface
+	{
+		restConfig, err := r.tenantRestConfigProvider.NewRestConfig(ctx, key.ClusterID(cluster), cluster.Spec.ControlPlaneEndpoint.String())
+		if err != nil {
+			return k8sClient, microerror.Mask(err)
+		}
+
+		k8sClient, err := k8sclient.NewClients(k8sclient.ClientsConfig{
```

Comment:
Just spot a bug here. You need to remove `:` from here. Otherwise the scope variable for success return won't be correct. Fix also error returns with `return nil, microerror.Mask(err)`.

Reply:
Fixing it here https://github.com/giantswarm/azure-operator/pull/999